### DISCUSSION TITLE
Make NEXTAUTH_URL name configurable

### DISCUSSION THREAD

**Discussion by arvindell** [nextauthjs/next-auth#4974] [Ideas]
### Description 📓

This feature would allow setting a custom name for the `NEXTAUTH_URL` variable. I need this feature because I use [direnv](https://direnv.net/) to pass environment variables to my project with two different NextAuth-enabled apps. 

- `packages`
    - `dashboard` <--  NEXTAUTH_URL=http://localhost:3001/api
    - `website` <--  NEXTAUTH_URL=http://localhost:3000/api

I need to send the NEXTAUTH_URL variable to each project separately because the auth logic changes between them.

### How to reproduce ☕️

1. Create a monorepo with two NextAuth apps
2. Use a single .env file for passing environment variables
3. See how both apps get the same value for `NEXTAUTH_URL`

### Contributing 🙌🏽

Yes, I am willing to help implement this feature in a PR

**Comment by balazsorban44** [comment]
I'm not sure I follow step 2. It's uncommon in Next.js apps to use the same .env file for different apps (environments?). You can define two different values for two different projects in their respective .env file. :thinking:

Could you clarify a bit more?

**Reply by arvindell** [reply]
My project setup is a monorepo with two Next.js projects (both using next-auth) and other services. Each of these apps currently has its own `.env` file. However, to improve DX in our team we want to switch to a single `.env` file across the whole repository (many, like `NEXT_PUBLIC_API_URL` are always the same between apps anyway).

Making `NEXTAUTH_URL` configurable would allow me to change the next-auth endpoint for the two applications that use the library. I'd end up having something like this


```
APP_1_NEXTAUTH_URL=http://localhost:3000/api
APP_2_NEXTAUTH_URL=http://localhost:3001/api
```

Does my use case make sense?

**Reply by arvindell** [reply]
Hello, @balazsorban44. Have you had the opportunity to review my use case?

**Comment by ferpinedo** [comment]
Doing this would be great for teams using monorepos.  In our project we have two apps: one for admin users and another one for customers, they both are built with Next.js and they both use NextAuth. We would love to change `NEXTHAUTH_URL` variable name to be able to have only one `.env` file (since most of the variables are the same between apps and sometimes is confusing managing multiple `.env` files).

**Comment by arvindell** [comment]
Hi @balazsorban44. What do you think about my use case here? Would appreciate some input, thanks
